src/core/prometheus.cc

```diff
@@ -632,6 +712,38 @@ future<> write_text_representation(output_stream<char>& out, const config& ctx,
     });
 }
 
+future<> write_protobuf_representation(output_stream<char>& out, const config& ctx, metric_family_range& m) {
+    return do_for_each(m, [&ctx, &out](metric_family& metric_family) mutable {
+        std::string s;
+        google::protobuf::io::StringOutputStream os(&s);
+
+        auto& name = metric_family.name();
+        pm::MetricFamily mtf;
+
+        mtf.set_name(ctx.prefix + "_" + name);
+        mtf.mutable_metric()->Reserve(metric_family.size());
+        metric_family.foreach_metric([&mtf, &ctx](auto value, auto value_info) {
+            fill_metric(mtf, value, value_info.id, ctx);
+        });
+        if (!write_delimited_to(mtf, &os)) {
+            seastar_logger.warn("Failed to write protobuf metrics");
+        }
+        return out.write(s);
+    });
+}
+
+bool is_accept_text(const std::string& accept) {
+    std::vector<std::string> strs;
+    boost::split(strs, accept, boost::is_any_of(","));
+    for (auto i : strs) {
+        boost::trim(i);
+        if (boost::starts_with(i, "application/vnd.google.protobuf;")) {
+            return false;
+        }
+    }
```

Comment:
This needs to be conditional on `--experimental-prometheus-protobuf` so that we don't accept it until the format is set in stone and until we're sure we satisfy it correctly.

Although, if it's exactly the same format as before, they can't change because it will break compatibility with old application written for older Prometheus. @amnonh will the negotiation be the same as with older Prometheus?

Reply:
do you mean, we need to add a runtime option to seastar to enable/disable this ? and this option is off by default?

Reply:
Yes. I'm trying to avoid this scenario:

1. we enable and ship this
2. Prometheus change the format since it's still experimental
3. applications built at step 1 are now broken

btw, how did you test this?

We should include a unit test, maybe there is a simple python client we can use.

Reply:
@avikivity I'll take care of all follow up patches including disable by default, I just wanted to start from the point we were before, (e.g. back to what we had).

Reply:
> btw, how did you test this?

@avikivity hi Avi. sorry for the latency. i didn't test this. i merely reverted 8ed9771ae98130d5adbccdaea4ecbac738170939. and resolved the conflicts to unblock Amnon. i don't have the bandwidth adding a test at this moment. if it's a must-have to unblock this change, @amnonh probably you could help add one? you can send a pull request to my repo, so we don't need to create a new pull request in order to preserve the existing comments in this pull requests.